packages/terra-form-input/src/terra-dev-site/doc/example/controlled/InputProgrammaticSet.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useRef } from 'react';
+import classNames from 'classnames/bind';
+import Input from 'terra-form-input';
+import styles from './InputProgrammaticSet.module.scss';
+
+const cx = classNames.bind(styles);
+
+const InputProgrammaticSet = () => {
+  const [text, setText] = useState('');
+  const inputRef = useRef(null);
+
+  const setTextClick = () => {
+    if (inputRef.current) {
+      inputRef.current.value = 'IpsumLorem';
+      const customEvent = new Event('input', {
+        bubbles: true,
+        cancelable: false,
+      });
+      inputRef.current.dispatchEvent(customEvent);
+    }
+  };
+
+  const handleChange = (event) => {
+    setText(event.target.value);
+  };
+
+  const handleInput = (event) => {
```

Comment:
nitpick - aren't these the same functions? we could rename this to something generic and reuse it for onChange and onInput. something like `onInputUpdates`